packages/protocol/scripts/sourcify-publish.ts

```diff
@@ -0,0 +1,92 @@
+/* tslint:disable no-console */
+import FormData from 'form-data'
+import fs from 'fs'
+import fetch from 'node-fetch'
+import path from 'path'
+import Web3 from 'web3'
+
+/*
+ * A script that reads the artifacts from the build/contracts directory and publish using the sourcify api.
+ *
+ * Expects the following flags:
+ *   network: The network for which artifacts should be
+ *
+ * Run using yarn run sourcify-publish, e.g.:
+ * yarn run sourcify-publish \
+ *   --network alfajores --buildArtifacts ./build/contracts --reportPath ./report.json
+ *
+ * report.json is generated by the build script with this format:
+ *  [{
+ *  "contract": "<Contract>Proxy",
+ *  "function": "_setImplementation",
+ *  "args": "<Contract>",
+ *    },{
+ *  "contract": "Registry",
+ *  "function": "setAddressFor",
+ *  "args": ["<Contract>", "<ContractProxy address>"]
+ *  }]
+ *
+ */
+
+const _buildTargets = {
+  network: undefined,
+  buildArtifacts: undefined,
+  reportPath: undefined,
+}
+
+async function main(buildTargets: typeof _buildTargets) {
+  const artifactBasePath = buildTargets.buildArtifacts || './build/contracts'
+  const artifactPaths = fs.readdirSync(artifactBasePath)
+  const reportPath = buildTargets.reportPath || './report.json'
+  const report = require(path.join(process.cwd(), reportPath))
+  const network = buildTargets.network
+
+  const web3 = new Web3('http://localhost:8545')
+
+  console.log('Uploading sources & metadata')
+  console.log('============================')
+  console.log(artifactPaths)
+
+  for (const _path of artifactPaths) {
+    const artifact = require(path.join(process.cwd(), artifactBasePath, _path))
+
+    console.log()
+    console.log(artifact.contractName)
+    console.log('-'.repeat(artifact.contractName.length))
+
+    const chainId = await web3.eth.getChainId()
+    const reportContract = report.filter((r) => r.contract === artifact.contractName)[0]
```

Comment:
This doesn't handle the case where a contract is not in the upgrade proposal. In general, I think it would make more sense to, above, rather than looping over artifacts, to loop over `report` items. 

Note that for the first type of proposal, if the contract is `<Contract>Proxy`, we should verify the code of `<Contract>` at address `args[0]`. For the second type of proposal, we should verify the code `<Contract>Proxy` at address `args[1]`.